### PR TITLE
Add final integration tests

### PR DESCRIPTION
**Pull request by kylacochrane**
This PR adds the final integration tests to the `gasnomenclature` pipeline. These tests focus on validating the input parameters for the various processes.

This PR also updates documentation:

- CITATIONS.md
- LICENSE
- CHNAGELOG.md
- output.md

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).

### CODE REVIEW

File: tests/pipelines/main_gm_threshold.nf.test [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stdout.contains("ERROR ~ '--pd_distm hamming' is set, but '--gm_thresholds 2,1,0.5' contains fractions."
+                                            + " Please either set '--pd_distm scaled' or remove fractions from distance thresholds.")
+        }
+    }
+
+    test("Test pipeline with single threshold set to 1") {
```
Comment (apetkau) [inline tests/pipelines/main_gm_threshold.nf.test:100]: Could you change the title of this test to indicate you are testing a failure condition? Could you also test an equivalent success condition (as in a case where the supplied threshold does equal the number of threshold columns)? Or is this tested elsewhere?

File: tests/pipelines/main_gm_threshold.nf.test [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert (workflow.stdout =~ /Error \[1.0\]  supplied thresholds do not equal the number of threshold columns in reference_clusters.txt/).find()
+        }
+    }
+
+    test("Test pipeline with threshold set to 1,0") {
```
Comment (apetkau) [inline tests/pipelines/main_gm_threshold.nf.test:118]: Similar comment here, could you change title of test to indicate it's for a failure condition and add an equivalent test that is successfull?

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks really great Kyla. Thanks so much for your work 😄 . A few comments for you.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,8 +5,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## In-development
 
-- Fixed nf-core tools linting failures introduced in version 2.12.1.
-- Added phac-nml prefix to nf-core config
+## 1.0.5 - 2024/06/17
```
[the remark this answers is not in the document]
Comment (kylacochrane) [reply CHANGELOG.md]: Oh that makes more sense haha - changed here: [cdbe9ef](https://github.com/phac-nml/gasnomenclature/pull/15/commits/cdbe9ef8b5216a38abaa77aa314d9ebb47a0c12f)

File: tests/pipelines/main_gm_threshold.nf.test (hunk 2 above)
Comment (kylacochrane) [reply tests/pipelines/main_gm_threshold.nf.test]: Done here: [c57ef6c](https://github.com/phac-nml/gasnomenclature/pull/15/commits/c57ef6c0f1a1817ec11e7782369a133e20e8adfd)

File: tests/pipelines/main_gm_threshold.nf.test (hunk 1 above)
Comment (kylacochrane) [reply tests/pipelines/main_gm_threshold.nf.test]: Done here: [c57ef6c](https://github.com/phac-nml/gasnomenclature/pull/15/commits/c57ef6c0f1a1817ec11e7782369a133e20e8adfd)

**Review by apetkau — APPROVED** [review]
Thanks so much Kyla. This is amazing work 😄

**Pull request merged**